float32 v_left
float32 v_right